### PR TITLE
CVS-100525 GPT-J-6b demo

### PR DESCRIPTION
**Pull request by dkalinowski**
Demo for tokenizer usage poc.

### CODE REVIEW

File: demos/gptj_causal_lm/python/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# GPT-J Causal Language Modeling Demo {#ovms_demo_gptj_causal_lm}
+
+### Introduction
+This demo illustrates usage of GPT-like models in OpenVINO™ Model Server. GPT-J 6B model used in this example can be found at [huggingface](https://huggingface.co/EleutherAI/gpt-j-6B) (~25GB). Steps below automate download and conversion steps to be able to load it using OpenVINO™. Example python client provided at the end of the document requests Model Server for the next word of the sentence until `EOS` (end of sequence) token is received. 
+
+
+### Download the model
+
+Prepare the environment:
+```bash
```
Comment (ngrozae) [inline demos/gptj_causal_lm/python/README.md:10]: git clone https://github.com/openvinotoolkit/model_server.git
cd model_server/demos/gptj_causal_lm/python

File: demos/gptj_causal_lm/python/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+### Convert the model
+The model needs to be converted to ONNX format in order to load in OVMS:
+```bash
+chmod +x convert_model.sh && ./convert_model.sh
+```
+The model will reside in `onnx/1` directory.
+
+The script should provide result confirming successful model conversion:
+```
```
Comment (ngrozae) [inline demos/gptj_causal_lm/python/README.md:31]: We can check this output:
```suggestion
```bash
```

File: demos/gptj_causal_lm/python/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+### Run the OVMS simple client script
+The script will validate the configuration and setup of OVMS.
+```bash
+python3 infer_ovms.py --url localhost:9000 --model_name gpt-j-6b
+```
+
+Desired output:
+```
```
Comment (ngrozae) [inline demos/gptj_causal_lm/python/README.md:57]: We can check this output:
```suggestion
```bash
```

File: demos/gptj_causal_lm/python/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+### Run the inference with pytorch
+We run the inference with pytorch to compare the result:
+```bash
+python3 infer_torch.py
+```
+
+Output:
+```
```
Comment (ngrozae) [inline demos/gptj_causal_lm/python/README.md:74]: We can check this output:
```suggestion
```bash
```

File: demos/gptj_causal_lm/python/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+### Interactive OVMS demo
+
+Run `app.py` script to run interactive demo predicting the next word in a loop until end of sentence token is encountered.
+
+```bash
+python3 app.py --input "Neurons are fascinating"
+```
+
+Output:
+```
```
Comment (ngrozae) [inline demos/gptj_causal_lm/python/README.md:94]: We can check this output:
```suggestion
```bash
```

File: demos/gptj_causal_lm/python/app.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    start_time = time.time()
+    results = client.predict(inputs=dict(inputs), model_name='resnet')
+    last_latency = time.time() - start_time
+    predicted_token_id = token = torch.argmax(torch.nn.functional.softmax(torch.Tensor(results[0,-1,:]),dim=-1),dim=-1)
+    word = tokenizer.decode(predicted_token_id)
+    input_sentence += word
+    os.system('cls' if os.name == 'nt' else 'clear')
+    print(f"Iteration: {iteration}\nLast predicted token: {predicted_token_id}\nLast latency: {last_latency}s\n{input_sentence}")
+    iteration += 1
+    if predicted_token_id == 198:
```
Comment (atobiszei) [inline demos/gptj_causal_lm/python/app.py:30]: Token id count could be parametrized?

File: demos/gptj_causal_lm/python/app.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import os
```
Comment (atobiszei) [inline demos/gptj_causal_lm/python/app.py:1]: Lack of license header

File: demos/gptj_causal_lm/python/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+transformers
+transformers[onnx]
+onnxruntime
+torch
+ovmsclient==2022.2
```
Comment (atobiszei) [inline demos/gptj_causal_lm/python/requirements.txt:5]:
```suggestion
ovmsclient==2022.2
```
Shouldn't we have here 2022.3? @mzegla

Comment (mzegla) [reply demos/gptj_causal_lm/python/requirements.txt:5]: ovmsclient 2022.3 is already on develop, but not yet published on PyPi. 
It's planned to publish along with the next minor release, so now pip install ovmsclient==2022.3 will not work yet. 
Perhaps we could drop specifying version in that case, so it's latest by default?

Comment (dkalinowski) [reply demos/gptj_causal_lm/python/requirements.txt:5]: Will drop specific version

**Review by atobiszei — APPROVED** [review]

**Review by ngrozae — APPROVED** [review]

File: demos/gptj_causal_lm/python/app.py (hunk 6 above)
Comment (dkalinowski) [reply demos/gptj_causal_lm/python/app.py]: done
